quic-api.md

```diff
@@ -0,0 +1,179 @@
+{
+    // Use QuicConnection
+}
+else
+{
+    // Fallback/Error
+}
+```
+Note that at the moment, both these properties are in sync and will report the same value, but that might change in the future. So we recommend to check [`QuicListener.IsSupported`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclistener?view=net-7.0) for server-scenarios and [`QuicConnection.IsSupported`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quicconnection.issupported?view=net-7.0) for the client ones.
+
+### QuicListener
+
+[`QuicListener`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclistener?view=net-7.0) represents a server side class that accepts incoming connections from the clients. The listener is constructed and started with a static method [`QuicListener.ListenAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclistener.listenasync?view=net-7.0). The method accepts an instance of [`QuicListenerOptions`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclisteneroptions?view=net-7.0) class with all the settings necessary to start the listener and accept incoming connections. After that, listener is ready to hand out connections via [`AcceptConnectionAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclistener.acceptconnectionasync?view=net-7.0). Connections returned by this method are always fully connected, meaning that the TLS handshake is finished and the connection is ready to be used. Finally to stop listening and release all resources, [`DisposeAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclistener.disposeasync?view=net-7.0) must be called.
+
+The sample usage of `QuicListener`:
+```C#
+using System.Net.Quic;
+
+// First, check if QUIC is supported.
+if (!QuicListener.IsSupported)
+{
+    Console.WriteLine("QUIC is not supported, check for presence of libmsquic and support of TLS 1.3.");
+    return;
+}
+
+// We want the same configuration for each incoming connection, so we prepare the connection option upfront and reuse them.
+// This represents the minimal configuration necessary to accept a connection.
+var serverConnectionOptions = new QuicServerConnectionOptions()
+{
+    // Used to abort stream if it's not properly closed by the user.
+    // See https://www.rfc-editor.org/rfc/rfc9000.html#name-application-protocol-error-
+    DefaultStreamErrorCode = 123,
+
+    // Used to close the connection if it's not done by the user.
+    // See https://www.rfc-editor.org/rfc/rfc9000.html#name-application-protocol-error-
+    DefaultCloseErrorCode = 456,
+
+    // Same options as for server side SslStream.
+    ServerAuthenticationOptions = new SslServerAuthenticationOptions
+    {
+        // List of supported application protocols, must be the same or subset of QuicListenerOptions.ApplicationProtocols.
+        ApplicationProtocols = new List<SslApplicationProtocol>() { SslApplicationProtocol.Http3 },
```

Comment:
I think we should use a different ALPN for example purposes, e.g. "quic-test"

Reply:
Aaaa, going another way and using other H/3 stuff for our samples. Personally, I prefer concrete, existing examples.

Reply:
Used a string constant.